PLNX-55: Upgrade ng-select and re-apply accessibility fixes

ShevonDenny opened the pull request:
Update ng-select to version 11.1.1 to be compatible with Angular 16

terencehonles committed:
build: update release.yml (#2247)

This change updates the release workflow to set the registry URL which
generates a config which allows the NODE_AUTH_TOKEN environmental
variable to be used, enables caching, and provides the NODE_AUTH_TOKEN
to only the publish step of the release workflow.

varnastadeus committed:
chore: update release.yml (#2252)

dependabot[bot] committed:
chore(deps): bump semver from 5.7.1 to 5.7.2 (#2256)

Bumps [semver](https://github.com/npm/node-semver) from 5.7.1 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.1...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump word-wrap from 1.2.3 to 1.2.4 (#2258)

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.4)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

miccehedin committed:
Make [items] take readonly array (#2219)

Tlepel committed:
feat: deselect option from dropdown in single select (#2175)

This feature enables deselecting an option from the
dropdown when `multiple` is `false`. The user can
click on a selected option, which will deselect that
option instead of closing the dropdown.

Fixes #2053

saibaburaavi committed:
fix(cicd): updates release.yml (#2265)

* Update release.yml

* Update release.yml

* Update release.yml

mahipureti committed:
fix: updates release.yml #2268

mahipureti committed:
fix: updates release.yml (#2269)

* Update release.yml

* Update release.yml

* Update release.yml

* Update release.yml

* Update release.yml

dependabot[bot] committed:
chore(deps): bump @babel/traverse from 7.21.5 to 7.23.2 (#2294)

Bumps [@babel/traverse](https://github.com/babel/babel/tree/HEAD/packages/babel-traverse) from 7.21.5 to 7.23.2.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.23.2/packages/babel-traverse)

---
updated-dependencies:
- dependency-name: "@babel/traverse"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

WaseemSabir committed:
Add Clear Button keyboard accessibility (#2245)

* Add Clear Button keyboard accessibility

* Add tests

* Fix linting

---------

Co-authored-by: Pavan Kumar Jadda <17564080+pavankjadda@users.noreply.github.com>

kheos31 committed:
Make select with [searchable]="false" focused on click and with keyboard (#2229)

https://github.com/ng-select/ng-select/issues/2221

Co-authored-by: Pavan Kumar Jadda <17564080+pavankjadda@users.noreply.github.com>

emir7 committed:
fix(ng-option): Remove event emits full ng-select model instead of direct value (#2289)

mahipureti committed:
chore: adds browser support in READMD.md and .browserslistrc file (#2297)

* docs: Update README.md

* chore: Update .browserslistrc

mahipureti committed:
feat: updates bootstrap 5 support and updates `@ng-bootstrap/ng-bootstrap` to v16 compatable version (#2299)

Co-authored-by: Pavan Kumar Jadda <17564080+pavankjadda@users.noreply.github.com>

ShevonDenny committed:
Update ng-select to version 11.1.1

PLNX-55

geebru commented on package.json:
```diff
@@ -1,70 +1,91 @@
+    "npm": ">= 8.19.4"
+  },
+  "scripts": {
+    "build": "ng build ng-select && ng build ng-option-highlight && yarn build:themes && yarn copy-sass",
+    "build:demo": "ng build demo --configuration production --base-href=/ng-select && yarn copy-examples",
+    "build:themes": "node-sass --output-style compressed src/ng-select/themes/ -o dist/ng-select/themes",
+    "copy-sass": "mkdir -p dist/ng-select/scss && cp src/ng-select/**/*.scss dist/ng-select/scss",
+    "copy-examples": "cp -r src/demo/app/examples dist/demo",
+    "start": "ng serve",
+    "test:watch": "ng test ng-select --watch",
+    "test": "ng test ng-select --code-coverage && ng test ng-option-highlight --code-coverage",
+    "test:ci": "ng test --watch=false --browsers=ChromeHeadless --code-coverage",
+    "coveralls": "cat ./coverage/ng-select/lcov.info | ./node_modules/coveralls/bin/coveralls.js",
+    "lint": "ng lint ng-select && ng lint ng-option-highlight",
+    "postpublish": "cp dist/ng-select/package.json src/ng-select && cp dist/ng-option-highlight/package.json src/ng-option-highlight",
+    "release": "semantic-release"
```
Keep the scripts from the previous version here - we don't use the same scripts they do.

ShevonDenny replied:
@geebru done.

geebru commented on package.json:
```diff
@@ -1,70 +1,91 @@
+    "postpublish": "cp dist/ng-select/package.json src/ng-select && cp dist/ng-option-highlight/package.json src/ng-option-highlight",
+    "release": "semantic-release"
+  },
+  "release": {
+    "branch": "master",
+    "plugins": [
+      "@semantic-release/commit-analyzer",
+      "@semantic-release/changelog",
+      "@semantic-release/github",
+      [
+        "@semantic-release/npm",
+        {
+          "pkgRoot": "dist/ng-select"
+        }
+      ],
+      "@semantic-release/release-notes-generator"
+    ]
+  },
```
We can nix this as well as we don't use their automated release strategy.

ShevonDenny replied:
@geebru done.

geebru reviewed (changes requested):
Two quick chore-related things to fix before merging.

ShevonDenny committed:
Update package.json file

ShevonDenny committed:
Merge commit '154c45d' into Angular16-Upgrade

ShevonDenny committed:
remove release.yml

geebru commented on src/ng-select/lib/ng-select.component.ts:
```diff
@@ -1,1105 +1,1134 @@
+            return;
+        }
+
+        if (this.deselectOnClick && item.selected) {
```
We use `multiple` in our Multi-Select Dropdowns, does this change impact that since it replaces the `multiple` check with `deselectOnClick` which we won't be adding at this time?

ShevonDenny replied:
@geebru I will test this.

ShevonDenny replied:
@geebru I don't see any breaking changes within the ng-select demo, I will test again with CAweb after the release candidate release

geebru commented on src/ng-select/lib/ng-select.component.ts:
```diff
@@ -1,1105 +1,1134 @@
+    }
+
+    private _onSelectionChanged() {
+        if (this.isOpen && this.deselectOnClick && this.appendTo) {
```
Same question here as the last review comment - does replacing `multiple` here break expected behavior in MSDs?

ShevonDenny replied:
@geebru I will test this.

ShevonDenny replied:
@geebru I don't see any breaking changes within the ng-select demo, I will test again with CAweb after the release candidate release

geebru reviewed (changes requested):
Noticed a swap on their end I wanted to check on before moving this forwards.

geebru commented on src/ng-select/lib/ng-select.component.html:
We need to put back the button here as it's an intentional accessibility improvement they're missing.

geebru replied:
This comment didn't show the changelog huh...

This references that we custom added a `<button>` replacement for the "Clear" action, and in the PR it looks like it reverted back to their `<span>`. We need to keep the button - and disregard their HTML.

ShevonDenny replied:
@geebru The button was added back

ShevonDenny committed:
Add back button that was removed with merge

PLNX-55

ShevonDenny committed:
Re-apply changes lost after merge

ShevonDenny committed:
Update background-color to previous color

Changed color for testing, reverted back to original color

geebru committed:
feat: Reorganize keycodes to dedicated method

PLNX-55

geebru committed:
feat: Revert back to our existing tab handler

Their updates create a focus trap in the clear button
which may be due to them using a span, while we use
a real button? Who knows. This works.

PLNX-55

geebru committed:
fix: Fix issue with MSD navigation attempting to work in SSD

Also resolve previous sibling erroring if there was no
previous sibling before wrapping to the last element

PLNX-55

geebru committed:
style: Fix spelling mistake

PLNX-55

geebru committed:
docs: Add Shevon as contributor

🎉

PLNX-55

geebru committed:
chore: Push lockfile changes

PLNX-55

geebru reviewed (approved):
Changes look good and the local demo seems to be consistent with expectations - thanks!
